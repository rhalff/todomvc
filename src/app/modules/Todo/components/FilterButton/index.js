// @flow
import React from 'react'
import styles from './scss/FilterButton.scss'

type Props = {
  active: boolean,
  name: string,
  title: string
}

const FilterButton = ({ active, name, title }: Props) => (
  <a
    name={name}
    href={`/#${name}`}
    className={active ? styles.active : styles.filterButton}
  >
    {title || name}
  </a>
)

export default FilterButton
