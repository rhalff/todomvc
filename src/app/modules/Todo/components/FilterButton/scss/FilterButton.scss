@import 'vars';

.filterButton {
  color: inherit;
  margin: 3px;
  padding: 3px 7px;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 3px;

  &:hover {
    border-color: $filterButtonHoverBorderColor;
  }
}

.active {
  composes: filterButton;
  border-color: $filterButtonActiveBorderColor;
}
