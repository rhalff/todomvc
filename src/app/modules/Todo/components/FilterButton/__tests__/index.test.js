// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import FilterButton from '../index'

test('Can show FilterButton', () => {
  const component = renderer.create(<FilterButton onChange={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Can show FilterButton (active)', () => {
  const component = renderer.create(
    <FilterButton onChange={() => {}} checked />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
