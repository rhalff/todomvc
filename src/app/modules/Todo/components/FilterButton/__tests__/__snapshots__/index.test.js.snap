// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show FilterButton (active) 1`] = `
<a
  className="filterButton"
  href="/#undefined"
  name={undefined}
/>
`;

exports[`Can show FilterButton 1`] = `
<a
  className="filterButton"
  href="/#undefined"
  name={undefined}
/>
`;
